lms/djangoapps/sga_api/views.py

```diff
@@ -0,0 +1,276 @@
+                },
+                status=status.HTTP_200_OK
+            )
+        except  Exception as e:
+            log.error(e.message)
+            return Response(
+                data={"error_message": e.message},
+                status=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            )
+
+    def __enroll_unenrol_users_for_courses(
+            self, courses_id, identifiers, request,
+            action='enroll', email_students=False, auto_enroll=False
+    ):
+        courses = {}
+        for course_id in courses_id:
+            results = self._enroll_unenrol_users_for_course(
+                course_id, identifiers, request,
+                action, email_students, auto_enroll
+            )
+            courses[course_id] = self._make_result_dict_for_course(
+                 results, action, auto_enroll
+            )
+        return courses
+
+    def _enroll_unenrol_users_for_course(
+            self, course_id, identifiers, request,
+            action='enroll', email_students=False, auto_enroll=False
+    ):
+        result_list = []
+        for identifier in identifiers:
+            try:
+                course_id_obj = SlashSeparatedCourseKey.from_deprecated_string(course_id)
+                course = get_course_by_id(course_id_obj)
+                if email_students:
+                    email_params = get_email_params(course, auto_enroll, secure=request.is_secure())
+                else:
+                    email_params = None
+
+                try:
+                    user = get_student_from_identifier(identifier)
+                    email = user.email
+                    language = get_user_email_language(user)
+                except User.DoesNotExist as e:
+                    if auto_enroll and action == 'enroll':
+                        email = identifier
+                        language = None
+                    else:
+                        raise e
+
+                validate_email(email)
+                if action == 'enroll':
+                    before, after, _ = enroll_email(
+                        course_id_obj, email, auto_enroll, email_students, email_params, language=language
+                    )
+                else:
+                    before, after = unenroll_email(
+                        course_id_obj, email, email_students, email_params, language=language
+                    )
+
+                result_list.append(
+                    self._make_result_dict_for_identifier(
+                        identifier=identifier, after=after.to_dict(), before=before.to_dict(),
+                    )
+                )
+            except User.DoesNotExist:
+                result_list.append(
+                    self._make_result_dict_for_identifier(
+                        identifier=identifier,
+                        error='User with {} identifier does not exist'.format(identifier)
+                    )
+                )
+            except Http404:
```

Comment:
That is not a good decision to use views function it would better to use `modulestore().get_course()`